src/bundle/Resources/public/js/scripts/udw/select.location.js

```diff
@@ -0,0 +1,131 @@
+    const udwContainer = doc.getElementById('react-udw');
+    const token = doc.querySelector('meta[name="CSRF-Token"]').content;
+    const siteaccess = doc.querySelector('meta[name="SiteAccess"]').content;
+    const findLocationsByIdList = (idList, callback) => {
+        const body = JSON.stringify({
+            ViewInput: {
+                identifier: `udw-locations-by-path-string-${idList.join('-')}`,
+                public: false,
+                LocationQuery: {
+                    Criteria: {},
+                    FacetBuilders: {},
+                    SortClauses: { SectionIdentifier: 'ascending' },
+                    Filter: { LocationIdCriterion: idList.join(',') },
+                    limit: 50,
+                    offset: 0,
+                },
+            },
+        });
+        const request = new Request('/api/ezp/v2/views', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.ez.api.View+json; version=1.1',
+                'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+                'X-Requested-With': 'XMLHttpRequest',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            body,
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
+        const errorMessage = Translator.trans(
+            /*@Desc("Cannot find children locations with given id - %idList%")*/ 'select_location.error',
+            { idList: idList.join(',') },
+            'universal_discovery_widget'
+        );
+
+        fetch(request)
+            .then(window.eZ.helpers.request.getJsonFromResponse)
+            .then((json) => callback(json))
+            .catch(() => window.eZ.helpers.notification.showErrorNotification(errorMessage));
+    };
+    const removeRootFromPathString = (pathString) => {
+        const pathArray = pathString.split('/').filter((val) => val);
+
+        return pathArray.splice(1, pathArray.length - 1);
+    };
+    const buildBreadcrumbsString = (viewData) => {
+        const searchHitList = viewData.View.Result.searchHits.searchHit;
+
+        return searchHitList.map((searchHit) => searchHit.value.Location.ContentInfo.Content.Name).join(' / ');
+    };
+    const toggleVisibility = (btn, pathString) => {
+        btn.hidden = !!pathString;
+
+        const contentNameWrapper = doc.querySelector(btn.dataset.contentBreadcrumbsSelector);
+
+        if (contentNameWrapper) {
+            contentNameWrapper.hidden = !pathString;
+        }
+    };
+    const setPath = (btn, pathString) => {
+        const pathStringInput = doc.querySelector(btn.dataset.locationPathInputSelector);
+
+        pathStringInput.value = pathString;
+        pathStringInput.dispatchEvent(new Event('change'));
+
+        const contentNameContainer = doc.querySelector(`${btn.dataset.contentBreadcrumbsSelector} div`);
```

Comment:
`contentBreadcrumbsContainer`

Reply:
The selector `div` is not a good one. Why this has no class?

Reply:
Because we decided not to create separate block (BEM terminology) for `select.location` component and I didn't want to add too much elements to `ez-filters` block. 
I can add some `ez-filters` class to this `div` but then I think we should change this selector:
`const btns = doc.querySelectorAll('.ez-btn--udw-select-location');`
What do you think?

Reply:
@tischsoic this is highly incorrect approach to use selectors like `.some-class div` in JS code without seriously good reason. Whenever you want to access a specific DOM node you should add a CSS class to it, to avoid selector chaining and CSS specificity.
This `div` belongs to some component, right? So it has some meaning. It is a part of something that can be named. Can you point me the place this `div` resides, maybe I can help you with constructing the selector properly?

Reply:
The `div` is here: https://github.com/ezsystems/ezplatform-admin-ui/pull/711/files#diff-f5eca6a947a5a0955802071e5723d9c5R53
It actually has a class, I didn't use it because it corresponds to the `ez-filters` which for me is not generic enough as we use this selector: `doc.querySelectorAll('.ez-btn--udw-select-location');` (https://github.com/ezsystems/ezplatform-admin-ui/pull/711/files#diff-490854292fbfb1577b3b5bbefa54df28R2) to search for "select location" buttons.

Should I use some other CSS class or add id to this div?

Reply:
Wait, are you trying to do some generic solution to be able to reuse this in other places? If so, this is a bad approach to achieve that.

You should create a class where you can override the selectors/methods on so.

Do we need this in other places? I think there is no point to spend time on this. When we will need this in other places we can change it (Once we will introduce this as a generic solution we will need to keep BC on this).

Reply:
I will just use this class from `div`.